visualizer/IdealGraphVisualizer/SourceRepository/src/main/java/org/graalvm/visualizer/source/impl/ui/LocationOpener.java

```diff
@@ -63,24 +67,48 @@ private void openOrView(boolean focus) {
             return;
         }
         int line = location.getLine();
+        int[] offsets = location.getOffsetsOrNull();
```

Comment:
Text based languages (like Truffle ones) are likely to fill offsets rather than line numbers. Passing the offsets to the `LocationOpener` in faba84a and using the offsets rather than line when present. More an experiment for your consideration than bullet proof code right now.

Reply:
I don't have much opinion since I never use this functionality so you should make it work in some rational way.  If the two ways of addressing the text disagree then it's unclear what you should do.  Naively I might assume that the line number is more to be trusted given that offsets seem a bit squishy given differences in line terminations between windows and unix, plus the lack of actual terminator characters in a UI element that displays text.  How are these mismatches expected to be dealt with?

Reply:
re. different newline terminators: the information about the OS the code was compiled in is lost during the compilation step, and cannot be even dumped to BGV. Maybe a hint for `truffle` to represent positions using line:column pair rather than offsets.

From the user's standpoint I agree that the whole line is too coarse; it's far better to point at a specific position, if we have such data.

I suggest to make *an option* for this: if the IGV knows (the user configures it) the line terminator use on the compiling machine, it can then recompute the offset to this machine's newline style and to line:column coordinates.

Reply:
I hope we'll find a way that works for current as well as future use-cases without a configuration option. I hope https://github.com/oracle/graal/pull/10510#issuecomment-2629232980 is a sound approach to achieve so.